modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteScrollableHitSource.java

```diff
@@ -176,9 +177,13 @@ public void onSuccess(org.elasticsearch.client.Response response) {
                             try (XContentParser xContentParser = xContentType.xContent().createParser(NamedXContentRegistry.EMPTY,
                                     content)) {
                                 parsedResponse = parser.apply(xContentParser, () -> ParseFieldMatcher.STRICT);
+                            } catch (ParsingException e) {
+                                /* Because we're streaming the response we can't get a copy of it here. The best we can do is hint that it
+                                 * is totally wrong and we're probably not talking to Elasticsearch. */
+                                throw new ElasticsearchException("Error parsing the response, remote is like not Elasticsearch.", e);
                             }
                         } catch (IOException e) {
-                            throw new ElasticsearchException("Error deserializing response", e);
+                            throw new ElasticsearchException("Error deserializing response, remote is likely not Elasticsearch", e);
```

Comment:
You're inconsistent in your use of periods (I prefer not having them but I don't mind either way).